ocean_lib/services/service.py

```diff
@@ -272,3 +299,73 @@ def update_compute_values(
         ] = trusted_algo_publishers
         self.compute_values["allowNetworkAccess"] = allow_network_access
         self.compute_values["allowRawAlgorithm"] = allow_raw_algorithm
+
+
+class ConsumerParameters:
+    def __init__(
+        self,
+        name: str,
+        type: Optional[str] = None,
+        label: Optional[str] = None,
+        required: Optional[bool] = False,
+        default: Optional[str] = None,
+        description: Optional[str] = None,
+        options: Optional[List[str]] = None,
+    ) -> None:
+        self.name = name
+        self.type = type
+        self.label = label
+        self.required = required
+        self.default = default
+        self.description = description
+
+        if options is not None and not isinstance(options, list):
+            raise TypeError("Options should be a list")
+
+        self.options = options
+
+    @classmethod
+    def from_dict(
+        cls, consumer_parameters_dict: Dict[str, Any]
+    ) -> "ConsumerParameters":
+        """Create a ConsumerParameters object from a JSON string."""
+        sd = copy.deepcopy(consumer_parameters_dict)
+        required = sd["required"] if "required" in sd else None
+        if required is not None:
+            required = (
+                bool(strtobool(sd["required"]))
+                if isinstance(sd["required"], str)
+                else required
+            )
+
+        return cls(
+            sd.pop("name", None),
+            sd.pop("type", None),
+            sd.pop("label", None),
+            required,
+            sd.pop("default", None),
+            sd.pop("description", None),
+            sd.pop("options", None),
+        )
+
+    @enforce_types
+    def as_dictionary(self) -> Dict[str, Any]:
+        """Return the consume parameters object as a python dictionary."""
+
+        result = {}
+        for attr_name in [
+            "name",
+            "type",
+            "label",
+            "required",
+            "default",
+            "description",
+            "options",
+        ]:
+            if getattr(self, attr_name) is not None:
+                if attr_name == "required":
+                    result["required"] = bool(self.required)
+
+                result[attr_name] = getattr(self, attr_name)
```

Comment:
This line will overwrite `result["required"]` so the previous if statement is pointless.

And even if this line wasn't here, I'd argue that casting `self.required` to a `bool` is overly defensive.  The type hint on `__init__` shows that it's `Optional[bool]` so if someone puts something other than `None`, `False`, or `True` in it, that's their problem.

So, I think we should remove the `if attr_name == "require":` block.

Reply:
I will recheck, but I added this guard because at some point `"required": None` was added to the list due to the way gettattr behaved. I'll come back with examples or correct this, thank you.

Reply:
I checked and you are right. I removed it. It was probably a remnant from some other draft logic I had previously.